drivers/ad5770r/ad5770r.c

```diff
@@ -0,0 +1,650 @@
+ */
+int32_t ad5770r_set_reference(struct ad5770r_dev *dev,
+			      bool external_reference,
+			      enum ad5770r_reference_voltage reference_selector)
+{
+	int32_t ret;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_REFERENCE,
+				    AD5770R_REFERENCE_RESISTOR_SEL(external_reference) |
+				    AD5770R_REFERENCE_VOLTATE_SEL(reference_selector));
+
+	if (ret)
+		return -1;
+
+	dev->reference_selector = reference_selector;
+	dev->external_reference = external_reference;
+
+	return ret;
+};
+
+/**
+ * Set reference configuration.
+ * @param dev - The device structure.
+ * @param alarm_config - Pointer to the array that contains the alarms
+ *			that will be enabled.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_alarm(struct ad5770r_dev *dev,
+			  bool * const alarm_config)
+{
+	int32_t ret;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_ALARM_CONFIG,
+				    AD5770R_ALARM_CONFIG_OPEN_DRAIN_EN(alarm_config[0]) |
+				    AD5770R_ALARM_CONFIG_THERMAL_SHUTDOWN_EN(alarm_config[1]) |
+				    AD5770R_ALARM_CONFIG_BACKGROUND_CRC_EN(alarm_config[2]) |
+				    AD5770R_ALARM_CONFIG_TEMP_WARNING_ALARM_MASK(alarm_config[3]) |
+				    AD5770R_ALARM_CONFIG_OVER_TEMP_ALARM_MASK(alarm_config[4]) |
+				    AD5770R_ALARM_CONFIG_NEGATIVE_CHANNEL0_ALARM_MASK(alarm_config[5]) |
+				    AD5770R_ALARM_CONFIG_IREF_FAULT_ALARM_MASK(alarm_config[6]) |
+				    AD5770R_ALARM_CONFIG_BACKGROUND_CRC_ALARM_MASK(alarm_config[7]));
+
+	if (ret)
+		return -1;
+
+	dev->alarm_config = alarm_config;
+
+	return ret;
+};
+
+/**
+ * Set the channel output filter resistor.
+ * @param dev - The device structure.
+ * @param output_filter - The structure that contains the channel output
+ *			filter resistor values.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_output_filter(struct ad5770r_dev *dev,
+				  enum ad5770r_output_filter_resistor output_filter,
+				  enum ad5770r_channels channel)
+{
+	return ad5770r_spi_reg_write(dev,
+				     AD5770R_OUTPUT_FILTER_CH0 + channel,
+				     AD5770R_OUTPUT_FILTER_CH(output_filter));
+};
+
+/**
+ * Set the hardware ldac configuration.
+ * @param dev - The device structure.
+ * @param mask_hw_ldac - The array contains HW LDAC channel masks.
+ * @return SUCCESS in case of success, negative error code otherwise.
+ */
+int32_t ad5770r_set_hw_ladc(struct ad5770r_dev *dev,
+			    bool * const mask_hw_ldac)
+{
+	int32_t ret;
+
+	ret = ad5770r_spi_reg_write(dev,
+				    AD5770R_HW_LDAC,
+				    AD5770R_HW_LDAC_MASK_CH(mask_hw_ldac[0], AD5770R_CH0) |
+				    AD5770R_HW_LDAC_MASK_CH(mask_hw_ldac[1], AD5770R_CH1) |
+				    AD5770R_HW_LDAC_MASK_CH(mask_hw_ldac[2], AD5770R_CH2) |
+				    AD5770R_HW_LDAC_MASK_CH(mask_hw_ldac[3], AD5770R_CH3) |
+				    AD5770R_HW_LDAC_MASK_CH(mask_hw_ldac[4], AD5770R_CH4) |
+				    AD5770R_HW_LDAC_MASK_CH(mask_hw_ldac[5], AD5770R_CH5));
+
+	if (ret)
+		return -1;
```

Comment:
maybe do,  
```
if (ret)
    return ret;
```
in all places;
that way we pass the error code up to the caller; so it doesn't get lost